subtests/docker_cli/liverestore/liverestore.py

```diff
@@ -0,0 +1,185 @@
+r"""
+Summary
+---------
+
+Test live-restore: running container should survive docker daemon restart.
+
+Operational Summary
+----------------------
+
+#. Run a docker container
+#. Restart docker daemon
+#. Verify that container is listed under 'docker ps' and is still running.
+
+Operational Detail
+------------------
+
+Our test container outputs a time value (seconds since epoch) every
+second. We check that it's running by running `docker logs` repeatedly
+until we get a new line; timing out at five seconds.
+
+We *restart* the docker daemon, not separate stop and then start.
+The latter would give us more testing options, such as confirming
+continued output from the container, but there's also a lot more
+possibility of leaving our system in an undefined state if something
+goes wrong between stop and start.
+
+Prerequisites
+-------------
+
+This test is only applicable if docker daemon is run with --live-restore
+option; we abort with TestNAError if it isn't.
+
+live-restore also requires `KillMode=process` in the docker systemd
+unit file. Without it, the container is stopped on daemon restart.
+See rhbz1424709#c40 and rhbz1460266.
+
+"""
+
+import time
+from autotest.client import utils
+import dockertest.docker_daemon as docker_daemon
+from dockertest import subtest
+from dockertest.containers import DockerContainers
+from dockertest.dockercmd import AsyncDockerCmd, DockerCmd
+from dockertest.images import DockerImage
+from dockertest.output.validate import mustpass
+from dockertest.xceptions import DockerTestNAError
+
+
+class liverestore(subtest.Subtest):
+
+    def initialize(self):
+        super(liverestore, self).initialize()
+
+        docker_cmdline = docker_daemon.cmdline()
+        self.failif_not_in("--live-restore", docker_cmdline,
+                           "docker daemon command-line options",
+                           DockerTestNAError)
+
+        # rhbz1424709#c40 : live-restore fails unless KillMode=process
+        killmode = docker_daemon.systemd_show('KillMode')
+        if killmode != 'process2':
+            self.logwarning("systemd KillMode is '%s' (expected 'process')."
+                            " This test will probably fail." % killmode)
+
+        self.stuff['dc'] = DockerContainers(self)
```

Comment:
This needs to be coordinated with https://github.com/cevich/autotest-docker-enabled-product-testing/pull/142 or similar/related otherwise it ain't gettin' tested in CI.